app/src/main/java/com/sdp/movemeet/utility/ActivitiesUpdater.java

```diff
@@ -1,173 +1,89 @@
-        else partcipantsId = (ArrayList<String>) participantsIdobj;
-
-        String sportobj = docSnap.getString("sport");
-        Sport sport;
-        if (sportobj == null) sport = Running;
-        else sport = Sport.valueOf(sportobj);
-        ; //enum sotred in firebase ?");
-
-        Activity act = new Activity(activityId, organizerId, title, numberParticipant, partcipantsId, longitude, latitude, description, documentPath, date, duration, sport, address, createdAt);
-        return act;
-    }
-
-    private String convertObjToString(Object obj) {
-        if (obj instanceof String && obj != null) {
-            return (String) obj;
-        } else return null;
-    }
-
-    private int convertLongToInt(Long obj) {
-        if (obj == null) return 0;
-        else return obj.intValue();
-    }
-
-    private double convertObjToDouble(Object obj) {
-        if (obj == null) return 0;
-        else return (double) obj;
-    }
-
-    private Date getDateObj(Date date) {
-        if (date == null) return new Date();
-        else return date;
-    }
-
-    private void addListeners(Query q, boolean updateMap) {
-        //Log.d(TAG, "cache allowed is" + cacheAllowed);
-        q.get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
-            @RequiresApi(api = Build.VERSION_CODES.N)
-            @Override
-            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
-                for (DocumentSnapshot docSnap : queryDocumentSnapshots.getDocuments()) {
-                    activities.add(convertDocSnapToActivity(docSnap));
-                    Log.d(TAG, "activities size is " + activities.size());
-                }
-                //if (cacheAllowed) cache.saveActivitiesInCache(activities);
-                if (updateMap && mapFragment.isAdded() && mapFragment.getGoogleMap() != null)
-                    mapFragment.displayNearbyMarkers();
-            }
-        }).addOnFailureListener(new OnFailureListener() {
-            @Override
-            public void onFailure(@NonNull Exception e) {
-                Log.d(TAG, e.toString());
+    /**
+     * On success, add the activities from db to the local list of activities
+     *
+     * @param task task containing the requested snapshots, that would then be deserialized into activities
+     */
+    private static Task addActivitiesOnSuccess(Task<QuerySnapshot> task) {
+        return task.addOnSuccessListener(queryDocumentSnapshots -> {
+            for (DocumentSnapshot docSnap : queryDocumentSnapshots.getDocuments()) {
+                activities.add(serializer.deserialize(docSnap.getData()));
+                Log.d(TAG, "activities size is " + activities.size());
             }
-        });
+        }).addOnFailureListener(e -> Log.d(TAG, e.toString()));
     }
 
-    public void clearLocalActivities() {
+    /**
+     * Clear the local list of activities
+     */
+    public static void clearLocalActivities() {
         activities.clear();
```

Comment:
Is this ".clear()" method also part of the abstraction? Because I'm not able to find it in the BackendManager.java interface...

Reply:
No, this is not part of the abstraction as it is used to clear the local activities, not the activities in the database (so it shouldn't be in the abstraction). But since for the moment it is only used once, I might remove it.

Reply:
Ok, I think I get it!